### PR TITLE
add Char#===(:Int8) casting, and vice versa

### PR DESCRIPTION
**Pull request by will**
The https://github.com/manastech/crystal/commit/35b177b1797c98d5852c1d2c0f17bcda0477b2ff patch made me think this might be a bit nicer for case statements on bytes.

### CODE REVIEW

File: src/char.cr [hunk 1]
```diff
@@ -496,4 +496,8 @@ struct Char
       io.write chars.to_slice, i
     end
   end
+
+  def ===(byte : UInt8)
+    self.ord === byte
```
Comment (jhass) [inline src/char.cr]: The `self.` should be redundant here.

Comment (will) [inline src/char.cr]: Good call

**Comment by asterite** [comment]
I think this makes sense, but sometimes I compare bytes with chars with `==` and in that case it won't work and maybe it will be a (bad) surprise.

One solution would be to consider bytes and chars as equal, but I'm also unsure about that.

**Comment by jhass** [comment]
`===` is the looser variant of `==`, I think `===` saying they're equal and `==` saying they aren't because they have different types makes perfect sense and is most certainly within my expectations as a Ruby developer.

**Comment by will** [comment]
Also, this doesn't include `UInt8#===(:Char)`, since that wasn't needed for the case statement to work.  I wonder if maybe it should be commutative though?

**Comment by MakeNowJust** [comment]
In non-ascii country people's (I am Japanese) opinion, `Char` is clearly not `UInt8`.  We want that `Char#===` should be matched to most of integer types because Unicode codepoint has 64bit integer size.

Why is it restricted `UInt8`?

**Comment by asterite** [comment]
We could make then `Char === Int` (all ints) and `Int === Char` give true if they have the same numeric value. Then we could do `97 === 'a'` and `'a' === 97` outside a case expression.

**Comment by will** [comment]
thanks @MakeNowJust and @asterite, I've repushed with all int support and made it commutative.

File: spec/std/char_spec.cr [hunk 2]
```diff
@@ -187,4 +187,14 @@ describe "Char" do
...
   end
+
+  it "#===(:Int)" do
+    ('c'.ord).should eq(99)
+    ('c' === 99_u8).should be_true
+    ('c' === 99).should be_true
+    ('z' === 99).should be_false
+
+    ('酒'.ord).should eq(37202)
+    ('酒' === 37202).should be_true
```
Comment (asterite) [inline spec/std/char_spec.cr]: Such a tasty and refreshing char ;-)

**Pull request merged**